### PR TITLE
✨ Adds interop delay option

### PR DESCRIPTION
**Pull request by tabshaikh**
**Description**

- by default the inclusiondelay is false 
- adds --interop.inclusiondelay flag in cli which delays sent messages by 5 seconds

Todo:

- [x] write tests 
- [x] update documentation

**Metadata**

#133

### CODE REVIEW

**Comment by jakim929** [comment]
discussed on tg: decided that it be better to include this logic in opsimulator rather than indexer

File: config/cli.go [hunk 1]
```diff
@@ -102,6 +103,12 @@ func ForkCLIFlags(envPrefix string) []cli.Flag {
 			Usage:   "enable interop predeploy and functionality",
 			EnvVars: opservice.PrefixEnvVar(envPrefix, "INTEROP_ENABLED"),
 		},
+		&cli.BoolFlag{
+			Name:    InteropInclusionDelayFlagName,
+			Value:   false, // enabled by default
+			Usage:   "enable interop inclusion delay functionality",
+			EnvVars: opservice.PrefixEnvVar(envPrefix, "INTEROP_INCLUSION_DELAY"),
+		},
```
Comment (jakim929) [inline config/cli.go:111]: should be a Uint64Flag

File: opsimulator/opsimulator.go [hunk 2]
```diff
@@ -411,6 +414,20 @@ func (opSim *OpSimulator) checkInteropInvariants(ctx context.Context, logs []typ
 			if common.BytesToHash(executingMessage.MsgHash[:]).Cmp(initiatingMsgPayloadHash) != 0 {
 				return fmt.Errorf("executing and initiating message fields are not equal")
 			}
+
+			if opSim.interopDelay != 0 {
+				// Add time check after getting the initiating message block header
+				executingBlockHeader, err := opSim.ethClient.HeaderByNumber(ctx, nil)
```
Comment (hamdiallam) [inline opsimulator/opsimulator.go:420]: s/executingBlockHeader/header

Lets make the variable names a bit less verbose, the executingBlock doesn't really mean anything here. We're just checking again the latest head of the chain

File: config/cli.go [hunk 3]
```diff
@@ -71,6 +72,12 @@ func BaseCLIFlags(envPrefix string) []cli.Flag {
 			Value:   "",
 			EnvVars: opservice.PrefixEnvVar(envPrefix, "LOGS_DIRECTORY"),
 		},
+		&cli.Uint64Flag{
+			Name:    InteropDelayFlagName,
+			Value:   0, // enabled by default
```
Comment (jakim929) [inline config/cli.go:77]: nit: remove comment

**Review by jakim929 — APPROVED** [review]
looks good! one nit for comment left but not super important. ready to go once CI passes

**Pull request merged**